internal/file/file.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package file
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+
+	digest "github.com/opencontainers/go-digest"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+// PrepareContent prepares the content descriptor from the file path or stdin.
+func PrepareContent(path string, mediaType string) (ocispec.Descriptor, io.ReadCloser, error) {
+	if path == "" {
+		return ocispec.Descriptor{}, nil, fmt.Errorf("missing file name")
+	}
+
+	if path == "-" {
+		content, err := io.ReadAll(os.Stdin)
+		if err != nil {
+			return ocispec.Descriptor{}, nil, err
+		}
+		rc := io.NopCloser(bytes.NewReader(content))
+
+		return ocispec.Descriptor{
+			MediaType: mediaType,
+			Digest:    digest.FromBytes(content),
+			Size:      int64(len(content)),
+		}, rc, nil
+	}
+
+	fp, err := os.Open(path)
+	if err != nil {
+		return ocispec.Descriptor{}, nil, fmt.Errorf("failed to open %s: %w", path, err)
+	}
+
+	fi, err := os.Stat(path)
```

Comment:
Call `Stat()` against `fp` to avoid potential race conditions in the filesystem.
```suggestion
	fi, err := fp.Stat()
```

Alternatively, we should be able to get the file size by `f.Seek(0, os.SEEK_CUR)` after `digest.FromReader(fp)`.

/cc @qweeah for comments.

Reply:
Should it be `f.Seek(0, os.SEEK_SET)`?

Reply:
Yes, we can use `fp.Seek(0, io.SeekCurrent)` to get the file size. Thanks for pointing this out.
ref:
```
// Seek whence values.
//
// Deprecated: Use io.SeekStart, io.SeekCurrent, and io.SeekEnd.
const (
	SEEK_SET int = 0 // seek relative to the origin of the file
	SEEK_CUR int = 1 // seek relative to the current offset
	SEEK_END int = 2 // seek relative to the end
)
```